packages/daemon/src/daemon.js

```diff
@@ -182,19 +183,43 @@ const makeDaemonCore = async (
 
   /**
    * The two functions "formulate" and "provide" share a responsibility for
-   * maintaining the memoization tables "controllerForId" and
+   * maintaining the memoization tables "controllerForId", "typeForId", and
    * "idForRef".
    * "formulate" is used for creating and persisting new formulas, whereas
    * "provide" is used for "reincarnating" the values of stored formulas.
    */
 
   /**
-   * Reverse look-up, for answering "what is my name for this near or far
-   * reference", and not for "what is my name for this promise".
+   * Forward look-up, for answering "what is the value of this id".
    * @type {Map<string, import('./types.js').Controller>}
    */
   const controllerForId = new Map();
 
+  /**
+   * Forward look-up, for answering "what is the formula type of this id".
+   * @type {Map<string, string>}
+   */
```

Comment:
Perhaps we can get a leg up on building a full formula memo here instead of just formula type, then use that anywhere we previously read formulas.

Reply:
#2165